src/rust/lib/system/web/src/platform.rs

```diff
@@ -0,0 +1,42 @@
+//! This module provides helpers for platform specific logic.
+
+use super::window;
+
+/// This enumeration lists all the supported platforms.
+#[derive(Debug,Clone,Copy)]
+pub enum Platform {
+    Linux,
+    Android,
+    Windows,
+    MacOS,
+    IOS,
+    Unknown
+}
+
+impl Platform {
+    /// Queries which platform we are on.
+    pub fn query() -> Self {
+        let window = window();
+        let navigator = window.navigator();
+        let platform  = navigator.platform().unwrap_or_else(|_| "Unknown".into());
+        let agent     = navigator.user_agent().unwrap_or_else(|_| "Unknown".into());
+
+        const MACOS_PLATFORMS   : [&str; 4] = ["Macintosh", "MacIntel", "MacPPC", "Mac68K"];
+        const WINDOWS_PLATFORMS : [&str; 4] = ["Win32", "Win64", "Windows", "WinCE"];
+        const IOS_PLATFORMS     : [&str; 3] = ["iPhone", "iPad", "iPod"];
```

Comment:
Let use HashSet for better performance like `let macos_platforms : HashSet<&str> = [...].iter().collect();` and then check using "contains".